processor/k8sprocessor/processor.go

```diff
@@ -92,56 +95,31 @@ func (kp *kubernetesprocessor) ProcessMetrics(ctx context.Context, md pdata.Metr
 func (kp *kubernetesprocessor) ProcessLogs(ctx context.Context, ld pdata.Logs) (pdata.Logs, error) {
 	rl := ld.ResourceLogs()
 	for i := 0; i < rl.Len(); i++ {
-		kp.processResource(ctx, rl.At(i).Resource(), k8sIPFromAttributes())
+		kp.processResource(ctx, rl.At(i).Resource())
 	}
 
 	return ld, nil
 }
 
-func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource, attributeExtractors ...ipExtractor) {
-	var podIP string
+func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource) {
 
-	for _, extractor := range attributeExtractors {
-		podIP = extractor(resource.Attributes())
-		if podIP != "" {
-			break
-		}
-	}
-
-	// Check if the receiver detected client IP.
-	if podIP == "" {
-		if c, ok := client.FromContext(ctx); ok {
-			podIP = c.IP
-		}
-	}
-
-	// If IP is still not available by this point, nothing can be tagged here. Return.
-	if podIP == "" {
+	podAttributes := k8sPodAssociationFromAttributes(ctx, resource.Attributes(), kp.podAssociations)
```

Comment:
We have another `attrsToAdd := kp.getAttributesForPod()` below and this one is named `podAttributes` as well. This could be a bit confusing to follow. IIRC, `podAttributes` here is actually a map of pod identifiers, such as IP, UID, etc, right? If so, let's name it `podIdentifiers` or something similar.